### PR TITLE
chore(Sounds): Started wiring notification sounds

### PR DESCRIPTION
**Pull request by InfamousVague**
Started wiring up the notification sounds but @sdwoodbury might have a better idea where the code should go (Ideally we can keep it out of components but we need access the app context to reference state.

### CODE REVIEW

File: ui/src/warp_runner/ui_adapter/message_event.rs [hunk 1]
```diff
@@ -43,6 +43,37 @@ pub async fn convert_message_event(
...
+                notifications_enabled && !state.read().ui.metadata.focused;
+
+            state.write().mutate(state::Action::AddNotification(
+                state::notifications::NotificaitonKind::Notification,
+                1,
+            ));
+
+            if should_dispatch_notification {
+                crate::utils::notifications::push_notification(
+                    "New friend request!".into(),
```
Comment (sdwoodbury) [inline ui/src/warp_runner/ui_adapter/message_event.rs]: needs translation

**Comment by InfamousVague** [comment]
This is almost ready, there is just a remaining bug where the sound is played twice when not focusing the app, and for some reason it's played even when the new message is coming from the active chat

**Comment by InfamousVague** [comment]
@phillsatellite this PR brings the ability to get notification sounds when you get a friend request and when messages come in.

File: ui/src/state/mod.rs [hunk 2]
```diff
@@ -771,7 +771,29 @@ impl State {
...
+                // Dispatch notifications only when we're not already focused on the application.
+                let notifications_enabled = self
+                    .configuration
+                    .config
+                    .notifications
+                    .friends_notifications;
+
+                if !self.ui.metadata.focused && notifications_enabled {
+                    crate::utils::notifications::push_notification(
+                        "New friend request!".into(),
```
Comment (sdwoodbury) [inline ui/src/state/mod.rs]: needs translations

File: ui/src/state/mod.rs [hunk 3]
```diff
@@ -833,6 +855,41 @@ impl State {
...
+
+                // This should be called if we have notifications enabled for new messages
+                if should_dispatch_notification {
+                    let sound = if self.configuration.config.audiovideo.message_sounds {
+                        Some(crate::utils::sounds::Sounds::Notification)
+                    } else {
+                        None
+                    };
+                    crate::utils::notifications::push_notification(
+                        "New friend request!".into(),
```
Comment (sdwoodbury) [inline ui/src/state/mod.rs]: same

**Review by sdwoodbury — CHANGES_REQUESTED** [review]

**Comment by phillsatellite** [comment]
@InfamousVague Yo! So I had a few issues,

This I replicated on both Windows and MacOS, it seems like _Message Sounds_ toggle is missing a click 
https://user-images.githubusercontent.com/93608357/217663577-b30cc504-a476-44ae-ada6-6165b9fdfbef.mov

The second issue @tooshel had on dev branch yesterday but I still havent been able to replicate on there. When I have a Windows User message a Mac User everytime the Mac User receives a message the text input box shoots to the top of the page

https://user-images.githubusercontent.com/93608357/217667045-b49334ae-f512-4066-a65f-1f98a6a2be6d.mov

**Comment by InfamousVague** [comment]
> @InfamousVague Yo! So I had a few issues,
> 
> This I replicated on both Windows and MacOS, it seems like _Message Sounds_ toggle is missing a click https://user-images.githubusercontent.com/93608357/217663577-b30cc504-a476-44ae-ada6-6165b9fdfbef.mov
> 
> The second issue @tooshel had on dev branch yesterday but I still havent been able to replicate on there. When I have a Windows User message a Mac User everytime the Mac User receives a message the text input box shoots to the top of the page
> edit.mov

This could be related to line endings but I don't think it's related to this ticket since it just sends out the notification arbitrarily when we get the associated events. 

I will wire up the sound now though!

**Comment by InfamousVague** [comment]
@phillsatellite should all be good to go now, I was able to hear the flip sound on my end but only in one direction but I did resolve a different bug around that area. If the thing doesn't make the sound twice for right now I wouldn't sweat it, we might just end up having to rethink some of the stuff in the sound util in the future if it becomes a problem.

File: ui/src/state/configuration.rs [hunk 4]
```diff
@@ -46,6 +46,11 @@ impl Configuration {
         let _ = self.config.save();
     }
 
+    pub fn set_message_sounds(&mut self, status: bool) {
+        self.config.audiovideo.message_sounds = status;
+        let _ = self.config.save();
```
Comment (sdwoodbury) [inline ui/src/state/configuration.rs:51]: should only save if `status != self.config.audiovideo.message_sounds`

**Review by sdwoodbury — DISMISSED** [review]
the requested change from last time wasn't addressed. only the English translation was added. Translations should be added for all languages in the `shared/locales` folder

**Comment by InfamousVague** [comment]
> the requested change from last time wasn't addressed. only the English translation was added. Translations should be added for all languages in the `shared/locales` folder

I don't think we should require this, it's a little outta scope and I want to avoid crappy google translate translations by non native speaking developers. It should fallback just fine to the english version pending proper translation.

File: ui/src/state/configuration.rs (hunk 4 above)
Comment (InfamousVague) [reply ui/src/state/configuration.rs:51]: Let's address this in a future PR as every configuration method currently doesn't use this logic and it's out of scope of this PR.

**Review by sdwoodbury — APPROVED** [review]
now it compiles

**Pull request merged**